src/compas/plugins.py

```diff
@@ -401,15 +414,15 @@ def try_import(self, module_name):
             If importable, it returns the imported module, otherwise ``None``.
         """
         module = None
-
         try:
             module = __import__(module_name, fromlist=["__name__"], level=0)
             self._cache[module_name] = True
 
         # There are two types of possible failure modes:
         # 1) cannot be imported, or
         # 2) is a python 3 module and we're in IPY, which causes a SyntaxError
-        except (ImportError, SyntaxError):
+        # 3) in Rhino8 we may get a nasty DotNet Exception when trying to import a module
+        except (ImportError, SyntaxError, DotNetException):
```

Comment:
does this mean, when not in Rhino, we accept `ImportError`, `SyntaxError` AND `BaseException`? does this mean we accept ALL exceptions?

Reply:
outside Rhino we would accept (in addition to the other two) a `DummyDotNetException` which is never really raised anywhere, so as good as ignored
Outside of Rhino any import error should always be one `ImportError` and `SyntaxError` anyways